src/language-js/utils/index.js

```diff
@@ -504,16 +504,26 @@ function templateLiteralHasNewLines(template) {
   return template.quasis.some((quasi) => quasi.value.raw.includes("\n"));
 }
 
+/**
+ * @param {Estree.TemplateLiteral | Estree.TaggedTemplateExpression} node
+ * @returns {boolean}
+ */
+function isTemplateWithLineBreaks(node) {
+  return (
+    (node.type === "TemplateLiteral" && templateLiteralHasNewLines(node)) ||
+    (node.type === "TaggedTemplateExpression" &&
+      templateLiteralHasNewLines(node.quasi))
+  );
```

Comment:
This part of logic is known problematic, please do not use.
Maybe we can check `expression`s instead.

Reply:
Can do!  Just to be clear, you're recommending checking whether the template has expressions _instead_ of checking if it has newlines?  (Because I would expect a single-line template like <code>\`foo bar\`</code> to wrap the same as a single line template with an interpolation, like <code>\`foo ${bar}\`</code>.)

Or are you talking about an API I can use to better check if a template has newlines?

Reply:
I'm not sure what are you going to achieve, just pointing out this existing function is not good to use.

Reply:
Ah, yeah, I hope to treat template literals a bit more like string literals.  For now, to reduce scope, I chose to focus on template literals that do not contain any new lines.

Any chance you could share links to some of the issues that have come from this problematic method so that I know the problems it causes?  That will help me understand how to avoid it better.